File: Data Structures For Interview/Java/Matrix Programs/Sparse.java
class Sparse {

    static void checkSparse(int m[][]) {

        int row = m.length;
        int col = m[0].length;

        int count = 0;

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                if (m[i][j] == 0) {
                    count++;
                }
            }
        }
        int sparse = (row * col) / 2;

        if (count > sparse) {
            System.out.println("The given matrix is Sparse Matrix");
        } else {
            System.out.println("The given matrix is not Sparse Matrix");
        }
    }

    public static void main(String s[]) {

        int m[][] = {
                { 1, 2, 3 },
                { 4, 5, 6 },
                { 7, 8, 9 }
        };

        int m2[][] = {
                { 4, 0, 0 },
                { 0, 5, 0 },
                { 0, 0, 6 }
        };

        checkSparse(m);
        checkSparse(m2);
    }
}

File: Data Structures For Interview/Python/Matrix Programs/Sparse.py
def sparse(list1):

    zero=0
    row=len(list1)
    print("Row :- ",row)
    col=len(list1[0])
    print("Col :- ",col)
    size=row*col
    print("Size :- ",size)
    print("Size :- ",size/2)

    for i in range(len(list1)):
        for j in range(len(list1[0])):
            if list1[i][j]==0:
               zero=zero+1
    if zero > (size/2):
        print("The given matrix is an sparse Matrix")
    else:
       print("The given matrix is not an sparse Matrix") 

list1=[
    [1,0,0],
    [0,1,0],
    [0,0,1]
]
list2=[
    [1,0,0],
    [1,1,1],
    [1,1,1]
]

sparse(list1)
sparse(list2)